cmd/thanos/rule.go

```diff
@@ -922,6 +923,12 @@ func queryFuncCreator(
 		}
 
 		return func(ctx context.Context, qs string, t time.Time) (promql.Vector, error) {
+			// See prometheus rules/group.go to see how this is populated
+			m := ctx.Value(promql.QueryOrigin{})
+			ruleGroup := m.(map[string]interface{})["ruleGroup"].(map[string]string)
```

Comment:
We could probably cast this to a concrete type?

Reply:
I have:
1. Cast the context value to map[string]interface{}:
`origin, ok := ctx.Value(promql.QueryOrigin{}).(map[string]interface{})`

2. Cast the rule group to map[string]string:
`ruleGroup, ok := origin["ruleGroup"].(map[string]string)`